src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller_TypeChar.cs

```diff
@@ -63,28 +57,24 @@ void ICommandHandler<TypeCharCommandArgs>.ExecuteCommand(TypeCharCommandArgs arg
                 return;
             }
 
-            // Separate the sig help providers into two buckets; one bucket for those that were triggered
-            // by the typed character, and those that weren't.  To keep our queries to a minimum, we first
-            // check with the textually triggered providers.  If none of those produced any sig help items
-            // then we query the other providers to see if they can produce anything viable.  This takes
-            // care of cases where the filtered set of providers didn't provide anything but one of the
-            // other providers could still be valid, but doesn't explicitly treat the typed character as
-            // a trigger character.
-            var filteredProviders = FilterProviders(allProviders, args.TypedChar);
-            var textuallyTriggeredProviders = filteredProviders.Item1;
-            var untriggeredProviders = filteredProviders.Item2;
-            var triggerInfo = new SignatureHelpTriggerInfo(SignatureHelpTriggerReason.TypeCharCommand, args.TypedChar);
+            var service = GetSignatureHelpService();
```

Comment:
this could return null.  Shoul this code check for that and deal with it accordingly?